src/api/api.ts

```diff
@@ -0,0 +1,537 @@
+  defaultRequest(
+    client,
+    `tasks?$filter=_regardingobjectid_value eq '${id}'`,
+    "GET"
+  );
+
+export const editCall = async (
+  client: IDeskproClient,
+  id: string,
+  data: ICreateCall
+): Promise<ProxyResponse> => {
+  return defaultRequest(client, `phonecalls(${id})`, "PATCH", data);
+};
+
+export const createCall = async (
+  client: IDeskproClient,
+  data: ICreateCall
+): Promise<ProxyResponse> => {
+  return defaultRequest(client, `phonecalls`, "POST", data);
+};
+
+export const getCallById = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: ICall[] }> =>
+  defaultRequest(client, `phonecalls?$filter=activityid eq '${id}'`, "GET");
+
+export const getFirst5CallsByContactId = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: ICall[] }> =>
+  defaultRequest(
+    client,
+    `phonecalls?$filter=_regardingobjectid_value eq '${id}'&$top=5`,
+    "GET"
+  );
+
+export const getCallsByContactId = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: ICall[] }> =>
+  defaultRequest(
+    client,
+    `phonecalls?$filter=_regardingobjectid_value eq '${id}'`,
+    "GET"
+  );
+
+export const editAppointment = async (
+  client: IDeskproClient,
+  id: string,
+  data: ICreateAppointment
+): Promise<ProxyResponse> => {
+  return defaultRequest(client, `appointments(${id})`, "PATCH", data);
+};
+
+export const createAppointment = async (
+  client: IDeskproClient,
+  data: ICreateAppointment
+): Promise<ProxyResponse> => {
+  return defaultRequest(client, `appointments`, "POST", data);
+};
+
+export const getAppointmentById = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: IAppointment[] }> =>
+  defaultRequest(client, `appointments?$filter=activityid eq '${id}'`, "GET");
+
+export const getFirst5AppointmentsByContactId = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: IAppointment[] }> =>
+  defaultRequest(
+    client,
+    `appointments?$filter=_regardingobjectid_value eq '${id}'&$top=5`,
+    "GET"
+  );
+
+export const getAppointmentsByContactId = (
+  client: IDeskproClient,
+  id: string
+): Promise<{ value: IAppointment[] }> =>
+  defaultRequest(
+    client,
+    `appointments?$filter=_regardingobjectid_value eq '${id}'`,
+    "GET"
+  );
+
+const isResponseError = (response: ProxyResponse) =>
+  response.status < 200 || response.status >= 400;
+
+const defaultRequest = async (
+  client: IDeskproClient,
+  endpoint: string,
+  method: RequestMethod,
+  data?: unknown
+) => {
+  const fetch = await proxyFetch(client);
+
+  const options: V2ProxyRequestInit = {
+    method,
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer [[oauth/global/access_token]]`,
+    },
+  };
+
+  if (data) {
+    options.body = JSON.stringify(data);
+  }
+
+  const url = `__client_url__${endpoint}`;
+
+  let response = await fetch(url, options);
+
+  if ([400, 401, 203, 500, 403].includes(response.status)) {
```

Comment:
I see exchange refresh token only from `global_access_token.json`. But what if we safe token oBut during of using the app, the access and refresh tokens are exchanged and saved in the user state.  And then I see that you save only the access token. 

And you should also try to exchange the access token with the refresh token from the user store (not only from `global_access_token.json`)

Reply:
no point in saving in the user state. they all use the exact same profile, set in the settings. We've discussed this flow before in dms and it is fine

Reply:
on line `510` - why do you save only the access token? refresh token won't be needed in the future?

Reply:
The default one works

Reply:
How long does an access token live?

Reply:
1 hour. the base refresh token works